clients/src/main/java/org/apache/kafka/clients/admin/Admin.java

```
@@ -1214,6 +1215,64 @@ default AlterClientQuotasResult alterClientQuotas(Collection<ClientQuotaAlterati
      */
     AlterClientQuotasResult alterClientQuotas(Collection<ClientQuotaAlteration> entries, AlterClientQuotasOptions options);
 
+    /**
+     * <p>Describe all SASL/SCRAM credentials.
+     *
+     * <p>This is a convenience method for {@link #describeUserScramCredentials(List, DescribeUserScramCredentialsOptions)}
+     *
+     * @return The DescribeUserScramCredentialsResult.
+     */
+    default DescribeUserScramCredentialsResult describeUserScramCredentials() {
+        return describeUserScramCredentials(null, new DescribeUserScramCredentialsOptions());
+    }
+
+    /**
+     * <p>Describe SASL/SCRAM credentials for the given users.
+     *
+     * <p>This is a convenience method for {@link #describeUserScramCredentials(List, DescribeUserScramCredentialsOptions)}
+     *
+     * @param users the users for which credentials are to be described; all users' credentials are described if null
+     *              or empty.  A user explicitly specified here that does not have a SCRAM credential will not appear
+     *              in the results.
```

Should we throw an exception for users which don't exist to be consistent with other APIs?

---

Hmm, good question.  The KIP doesn't state what do do here.  @cmccabe thoughts?

---

I originally was thinking we could just omit them.  But thinking about it more, I think we should have an error code to be consistent with our other APIs.  We'll need a new error code like USER_NOT_FOUND.  And we'll also need to add a per-user error code-- something we don't currently have.  But it shouldn't be too hard to do.

---

(Another advantage to having a per-user error code is that in the future, we may have more reasons why describing a user might fail, and these might be per-user)

---

@rondagostino : It still says "A user explicitly specified here that does not have a SCRAM credential will not appear in the results".  I thought we agreed that such a user would get an error code?

---

I reworked this Javadoc to list all possible exceptions for both alter and describe.